sale_rental/tests/test_sale_rental.py

```diff
@@ -0,0 +1,77 @@
+from odoo.exceptions import UserError
+from odoo.tests import Form
+from odoo.tests.common import TransactionCase
+
+
+class TestSaleRental(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.test_rental_prod = cls.env.ref("sale_rental.rent_product_product_25")
+        cls.test_partner = cls.env["res.partner"].create({"name": "Foo"})
+        cls.rental_in_loc = cls.env.ref("stock.warehouse0").rental_in_location_id
+        cls.rental_out_loc = cls.env.ref("stock.warehouse0").rental_out_location_id
+
+    def test_main(self):
+        # Rent a product
+        so_form = Form(self.env["sale.order"])
+        so_form.partner_id = self.test_partner
+        line_vals = {
+            "product_id": self.test_rental_prod.id,
+            "start_date": "2022-01-01",
+            "end_date": "2022-01-10",
+            "rental_qty": 1,
+        }
+
+        # with so_form.order_line.new() as line:
+        #     line.product_id = self.test_rental_prod
+        # line.start_date = "2022-01-01"
+        # line.end_date = "2022-01-10"
+        # line.rental_qty = 1
+        so = so_form.save()
+        so.write({"order_line": [(0, 0, line_vals)]})
+        sol = so.order_line
```

Comment:
In order to play the onchange
```suggestion
        with so_form.order_line.new() as line:
            line.product_id = self.test_rental_prod
            line.start_date = "2022-01-01"
            line.end_date = "2022-01-10"
            line.rental_qty = 1
            line.product_uom_qty = 10
        so = so_form.save()
        sol = so.order_line
```